### ISSUE TITLE
No further splits with positive gain, best gain: -inf

### ISSUE THREAD

**Issue by ninenerd** [microsoft/LightGBM#3440]
I am new to Lightgbm and there are few documentation, so seeking some help.

1.What does this mean while training ? (No further splits with positive gain, best gain: -inf). An error ?

2. I am training a regression model with Lightgbm, what should be the metric for parameter ? see below hyper_params. 

3. While fitting the model, there is something called eval_metric , should I use another metric there ?

Example :
I have over million records and predicting continous values.

```
hyper_params = {
    'task': 'train',
    'boosting_type': 'gbdt',
    'objective': 'regression',
    'metric': ['l2''],
    'learning_rate': 0.005,
    'feature_fraction': 0.9,
    'bagging_fraction': 0.7,
    'bagging_freq': 10,
    'verbose': 0,
    "max_depth": 8,
    "num_leaves": 128,  
    "max_bin": 512,
    "num_iterations": 100000,
    "n_estimators": 1000
}

gbm = lgb.LGBMRegressor(**hyper_params)


gbm.fit(X_train, y_train,
        eval_set=[(X_test, y_test)],
        eval_metric='l1',
        early_stopping_rounds=1000)
```

**Comment by guolinke** [comment]
Hi @ninenerd 
1. It is a warning,  not an error. It usually indicates that your setting of complexity is too high, or your data is not easy to fit. 
2. both `l1` and `l2` could be used in the regression. and you can set it both in `hyper_params['metric']`, and `eval_metric`. 
3. not need to set them both, they are the same parameter. LightGBM will show warnings when you set both of them.
